internal/gke/update.go

```diff
@@ -373,22 +375,37 @@ func UpdateLabels(
 	if config.Spec.Labels == nil || reflect.DeepEqual(config.Spec.Labels, upstreamSpec.Labels) || (upstreamSpec.Labels == nil && len(config.Spec.Labels) == 0) {
 		return NotChanged, nil
 	}
-	cluster, err := GetCluster(ctx, client, &config.Spec)
-	if err != nil {
-		return NotChanged, err
-	}
 	logrus.Infof("updating cluster labels for cluster [%s]", config.Name)
-	_, err = client.Projects.
-		Locations.
-		Clusters.
-		SetResourceLabels(
-			ClusterRRN(config.Spec.ProjectID, Location(config.Spec.Region, config.Spec.Zone), config.Spec.ClusterName),
-			&gkeapi.SetLabelsRequest{
-				LabelFingerprint: cluster.LabelFingerprint,
-				ResourceLabels:   config.Spec.Labels,
-			},
-		).Context(ctx).
-		Do()
+	backoff := wait.Backoff{
+		Duration: 5 * time.Second,
+		Steps:    2,
+	}
+	err := wait.ExponentialBackoff(backoff, func() (bool, error) {
+		cluster, err := GetCluster(ctx, client, &config.Spec)
+		if err != nil {
+			return false, err
+		}
+		_, err = client.Projects.
+			Locations.
+			Clusters.
+			SetResourceLabels(
+				ClusterRRN(config.Spec.ProjectID, Location(config.Spec.Region, config.Spec.Zone), config.Spec.ClusterName),
+				&gkeapi.SetLabelsRequest{
+					LabelFingerprint: cluster.LabelFingerprint,
+					ResourceLabels:   config.Spec.Labels,
+				},
+			).Context(ctx).
+			Do()
+		if err != nil && strings.Contains(err.Error(), "Labels could not be set due to fingerprint mismatch") {
```

Comment:
The fingerprint would always protect us from reconcile happening to quickly ?

Reply:
The fingerprint is a requirement from the GKE API, and this error string is the error it returns if the one we're providing is out of date. It doesn't affect the reconcile loop delay.